### PR TITLE
improve speed of cascade walls, better description for them + CL for cascade antag

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Cascade walls were processing on object subsystem, they are now in their own subsystem that ticks once per second and should be more reliable even in case of high td
better description for the walls to be more interesting

## Why It's Good For The Game
Proper performances for the crystals spread and better description for them

## Changelog

:cl:
qol: improved the speed of cascade wall spreading, made a better description for them
qol: I didn't update the changelog in the initial PR so i'll add this here: Antags can unlock a final objective (like romerol) that instructs them to cause a cascade. The objective will provide a destabilizing crystal that once applied to the SM will instantly cause it to start delaminating and sets the SM for a cascade. Once it delaminates it ensures the start of the cascade. NOTE that if the SM recovers you won't be able to do it again.
/:cl:

### CODE REVIEW

File: code/modules/power/supermatter/supermatter_cascade_components.dm [hunk 1]
```diff
@@ -1,6 +1,11 @@
+PROCESSING_SUBSYSTEM_DEF(supermatter_cascade)
+	name = "Supermatter Cascade"
+	wait = 1 SECONDS
+	stat_tag = "SC"
+
 /turf/closed/indestructible/supermatter_wall
-	name = "wall"
-	desc = "Effectively impervious any methods of destruction."
+	name = "crystal mass"
+	desc = "You see this massive crystal mass looming towards you, cracking and screeching at every seemingly random movement. Despair because the end is here."
```
Comment (tralezab) [inline code/modules/power/supermatter/supermatter_cascade_components.dm:8]: the last sentence comes off weird

File: code/modules/power/supermatter/supermatter_cascade_components.dm [hunk 2]
```diff
@@ -1,6 +1,11 @@
+PROCESSING_SUBSYSTEM_DEF(supermatter_cascade)
+	name = "Supermatter Cascade"
+	wait = 1 SECONDS
+	stat_tag = "SC"
```
Comment (tralezab) [inline code/modules/power/supermatter/supermatter_cascade_components.dm:4]: in it's own subsystem file, not here!

**Review by tralezab — CHANGES_REQUESTED** [review]
bro you can't just define a subsystem in your random components file like bro that's madness

**Comment by LemonInTheDark** [comment]
What does performance mean here? how does this make things faster?

**Comment by LemonInTheDark** [comment]
How much of that was the cost of doing the spreading?

**Comment by Ghilker** [comment]
> How much of that was the cost of doing the spreading?

the spreading keeps on expanding an already big list and i think that's the main cause, SSobj had a current run list of over 2000 elements

File: code/modules/power/supermatter/supermatter_cascade_components.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/power/supermatter/supermatter_cascade_components.dm]: i'll remove it

**Comment by LemonInTheDark** [comment]
You stop processing when there's no more spreading for a tile to do right?

**Pull request merged**